packages/rs-dapi-client/src/address_list.rs

```diff
@@ -198,6 +198,11 @@ impl AddressList {
     pub fn is_empty(&self) -> bool {
         self.addresses.is_empty()
     }
+
+    /// Getter function that returns a clone of the Hashset of addresses
+    pub fn addresses(&self) -> HashSet<Address> {
+        self.addresses.clone()
```

Comment:
I don't think it's a good idea to clone every call. Better to return a reference and user decide if he needs to clone it